### PR TITLE
Remove copy constructors from request classes and TransportMessage type

### PR DESCRIPTION
**Pull request by javanna**
As a followup of #15776 we can remove all copy constructors that are left in our request classes, that were used to copy headers and context. Also, `TransportMessage` doesn't need a generic type anymore.

### CODE REVIEW

**Comment by nik9000** [comment]
> Also, TransportMessage doesn't need a generic type anymore.

Hurray!

File: core/src/main/java/org/elasticsearch/action/index/IndexRequest.java [hunk 1]
```diff
@@ -160,26 +160,6 @@ public IndexRequest() {
     }
 
     /**
-     * Copy constructor that creates a new index request that is a copy of the one provided as an argument.
-     * The new request will inherit though headers and context from the original request that caused it.
-     */
-    public IndexRequest(IndexRequest indexRequest) {
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: I think I use this in reindex to use one index request to stamp out others. It seemed like a good idea at the time.

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: I don't think that is an argument for keeping this though - maybe an argument for reindex working another way.

File: core/src/main/java/org/elasticsearch/transport/TransportMessage.java [hunk 2]
```diff
@@ -25,11 +25,8 @@
 import org.elasticsearch.common.transport.TransportAddress;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
 
-public abstract class TransportMessage<TM extends TransportMessage<TM>> implements Streamable {
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/transport/TransportMessage.java]: Hurray!

**Comment by nik9000** [comment]
Does removing the generic-ness just fix 123412421 raw type errors? I'm a bit surprised everything still compiles but ok.

**Comment by nik9000** [comment]
LGTM I think.

File: core/src/main/java/org/elasticsearch/action/index/IndexRequest.java (hunk 1 above)
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: It's annoying how these things are error-prone, like you add a new instance member and you forget to add it to the copy constructor (just happened with pipeline, which triggered this whole PR). Would be nice to find another way :)

**Comment by s1monw** [comment]
cool thx @javanna

**Pull request merged**